lib/simulator/simulatormodule.c

```diff
@@ -952,7 +952,25 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
     gstate = TAKE_GIL();
 
     pSihHdl = PyTuple_GetItem(args, 0);
-    hdl = (gpi_sim_hdl)PyLong_AsUnsignedLong(pSihHdl);
```

Comment:
I observed the problem on a 64bit Debian host running Riviera-PRO 32bit version with Python 2.7.12   32bit version. In this setup, PyLong_AsUnsignedLong returns an invalid value (0xFFFFFFFF, I think). The reason for this is that the type of pSihHdl is actually a PyInt object, not a PyLong object. Calling the correct type conversion function returns the expected pointer value.

I tested also with 32bit Python 3.5. Here there is no problem. There is no PyInt type in Python 3.5, all integers are implemented as PyLong.

However, it is possible that some other code elsewhere is at fault by delivering a PyInt instead of a PyLong.